From 8b857741536708450a5a65a7ba05bbda723b8b7d Mon Sep 17 00:00:00 2001
From: Brent Stapleton <brent.stapleton@ettus.com>
Date: Thu, 22 Feb 2018 11:06:23 -0800
Subject: [PATCH 06/40] mpm: break up device compilation

Breaking up dboard compilation based on the target MPM_DEVICE.
---
 mpm/lib/CMakeLists.txt         | 5 ++++-
 mpm/lib/dboards/CMakeLists.txt | 9 +++++----
 mpm/tools/CMakeLists.txt       | 6 +++---
 3 files changed, 12 insertions(+), 8 deletions(-)

diff --git a/mpm/lib/CMakeLists.txt b/mpm/lib/CMakeLists.txt
index 684a4d1e3..091c2c25f 100644
--- a/mpm/lib/CMakeLists.txt
+++ b/mpm/lib/CMakeLists.txt
@@ -8,10 +8,13 @@ SET(UHD_HOST_ROOT ${CMAKE_SOURCE_DIR}/../host)
 
 ADD_SUBDIRECTORY(dboards)
 ADD_SUBDIRECTORY(chips)
-ADD_SUBDIRECTORY(mykonos)
 ADD_SUBDIRECTORY(spi)
 ADD_SUBDIRECTORY(types)
 
+if(MPM_DEVICE STREQUAL "n3xx")
+    ADD_SUBDIRECTORY(mykonos)
+endif(MPM_DEVICE STREQUAL "n3xx")
+
 USRP_PERIPHS_ADD_OBJECT(periphs
     exception.cpp
     xbar_iface.cpp
diff --git a/mpm/lib/dboards/CMakeLists.txt b/mpm/lib/dboards/CMakeLists.txt
index 7f728a836..a1017d8d3 100644
--- a/mpm/lib/dboards/CMakeLists.txt
+++ b/mpm/lib/dboards/CMakeLists.txt
@@ -8,7 +8,8 @@
 # This file included, use CMake directory variables
 ########################################################################
 
-USRP_PERIPHS_ADD_OBJECT(dboards
-  magnesium_manager.cpp
-)
-
+if(MPM_DEVICE STREQUAL "n3xx")
+    USRP_PERIPHS_ADD_OBJECT(dboards
+        magnesium_manager.cpp
+    )
+endif(MPM_DEVICE STREQUAL "n3xx")
diff --git a/mpm/tools/CMakeLists.txt b/mpm/tools/CMakeLists.txt
index 727635a02..1aa079f28 100644
--- a/mpm/tools/CMakeLists.txt
+++ b/mpm/tools/CMakeLists.txt
@@ -13,8 +13,8 @@ INSTALL(PROGRAMS
 SET(eeprom_tool_sources)
 SET(eeprom_tool_libs)
 
-IF("${MPM_DEVICE}" STREQUAL "n310")
-    MESSAGE(STATUS "Adding N310-specific EEPROM tools...")
+IF("${MPM_DEVICE}" STREQUAL "n3xx")
+    MESSAGE(STATUS "Adding N3XX-specific EEPROM tools...")
     SET(eeprom_tool_libs eeprom.c)
     SET(eeprom_tool_sources
         db-dump.c
@@ -27,7 +27,7 @@ IF("${MPM_DEVICE}" STREQUAL "n310")
         eeprom-set-flags.c
         fan-limits.c
     )
-ENDIF("${MPM_DEVICE}" STREQUAL "n310")
+ENDIF("${MPM_DEVICE}" STREQUAL "n3xx")
 
 #for each source: build an executable and install
 FOREACH(eeprom_tool_source ${eeprom_tool_sources})
-- 
2.11.0

